frontend/src/app/api/attachment-download/[opportunityId]/route.ts

```diff
@@ -0,0 +1,81 @@
+import { randomUUID } from "crypto";
+import * as zip from "@zip.js/zip.js";
+import { ApiRequestError, readError } from "src/errors";
+import { getOpportunityDetails } from "src/services/fetch/fetchers/opportunityFetcher";
+
+import { NextRequest, NextResponse } from "next/server";
+
+export async function GET(
+  request: NextRequest,
+  { params }: { params: Promise<{ opportunityId: string }> },
+): Promise<Response> {
+  const { opportunityId } = await params;
+  try {
+    const response = await getOpportunityDetails(opportunityId);
+
+    if (!response || response.status_code !== 200) {
+      throw new ApiRequestError(
+        `Error fetching Opportunity: ${response.message}`,
+        "APIRequestError",
+        response.status_code,
+      );
+    }
+
+    if (response.data.attachments?.length === 0) {
+      return Response.json(
+        {
+          message: `No files for Opportunity`,
+          opportunityId,
+        },
+        { status: 404 },
+      );
+    }
+    const zipWriter = new zip.ZipWriterStream();
+
+    const closeWriter = async () => {
+      await zipWriter.close();
+    };
+
+    Promise.allSettled(
+      response.data.attachments?.map((attachment) => {
+        return new zip.HttpReader(attachment.download_path).readable
+          .pipeTo(zipWriter.writable(attachment.file_name))
+
+          .catch((e: Error) => {
+            const uuid = randomUUID();
+            console.error(
```

Comment:
this will log in the catch block - might make more sense to just add whatever we want to log into the error message and have it log below?